sqlobject/src/main/java/org/jdbi/v3/sqlobject/WithHandleMethodHandlerFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import org.jdbi.v3.core.HandleCallback;
+import org.jdbi.v3.core.extension.ExtensionMethod;
+
+class WithHandleMethodHandlerFactory implements HandlerFactory {
+    private static final Method WITH_HANDLE = Handlers.methodLookup(SqlObject.class, "withHandle", HandleCallback.class);
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Optional<Handler> buildHandler(Class<?> sqlObjectType, Method method) {
+        if (!WITH_HANDLE.equals(method)) {
+            return Optional.empty();
+        }
+        return Optional.of((t, a, h) ->
+            h.invokeInContext(
```

Comment:
`Transactional.commit` and such still use `getHandle()`, so they'll keep throwing the "connection closed" exception. Should we maybe do something about that? `onDemand.commit` makes no sense to write in the first place, but it could maybe detect being invoked on an ondemand and throw a more meaningful exception since that's what this PR is about?

Maybe we need an OnDemand marker interface, since there was also that `instanceof` check on the handlesupplier to check for ondemand instances.

Reply:
So I tried using an OnDemand marker interface, it can't be done because the proxy with the interface on it isn't passed to the handler.

Reply:
I made it implement *all* declared interfaces in addition to your hard-coded `SqlObject`.